### ISSUE TITLE
Cursor snaps back to original position after dragging event in certain applications.

### ISSUE THREAD

**Issue by genativ** [hyprwm/Hyprland#1511]
### **The Issue**
In certain applications, my cursor resets it's position after dragging on the screen. On further inspection, it appears to only occur in applications where some kind of cursor constraints are applied. 

https://user-images.githubusercontent.com/8363722/217341660-d868c888-a2b0-4694-a8bf-b1fca261d22a.mp4

**Findings**
- At first I thought this was an issue exclusively to blender. It happened after dragging panels and moving objects in the 3d viewport. However, the same issue occurs in VMWare player when clicking and dragging anywhere.
- I know blender constrains the cursor with wayland support ([this](https://projects.blender.org/blender/blender/issues/98714) issue seems very similar). VMware Player also seems like it expects to lock a cursor to the VM.
- Before making this issue, I tried to reproduce the problem with Gnome DE (both Wayland and Xorg) and even another wlroots based compositor (Sway), but to no avail. That's why I believe this issue may be specific to Hyprland.
- With blender, this problem only occurs when running in Wayland mode. (sidenote: this also causes blender to open in a "fake fullscreen mode", where it seems to ignore all windows and just open across the screen. Only after opening the preferences window, blender returns to being regularly tiled). When opening blender with xwayland, this issue does not occur. With VMWare player the issue occurs, even though it's run with xwayland.
- I DO use Hyprland-nvidia-git with an RTX 3080, if that matters. I installed the nvidia-open-dkms on the regular linux kernal for Arch linux 6.1.9-arch1-2. 

---
**Steps to reproduce**
- Install Blender - or possibly another application with cursor constraints(?)
- press mouse button down and move the cursor
- release mouse button

**Expected Outcome**
- Cursor position after release should be the same as the cursor position on release

**Noted Outcome**
- Cursor position returns to the position the cursor was at when the mouse was clicked.

**Comment by vaxerski** [comment]
hm, thought I fixed that one.

**Comment by RochaLAJ** [comment]
Same here. Specially when i play MMORPGs. i tought we had fixed this one. I tried to use different mouses and even USB port.

Config file
```
general {
    max_fps = 150
    gaps_in = 2
    gaps_out = 20
    border_size = 3.2
    col.active_border = 0xffb072d1
    col.inactive_border = 0xff292a37
    layout = dwindle
}

decoration {
    rounding = 7
    blur = 0
    drop_shadow = 0
    shadow_range = 60
    shadow_render_power = 3
    col.shadow=0x66000000
}


animations {
    enabled=1
    speed=7
    windows_speed=6
    windows_curve=smooth
    windows=1
    borders=1
    borders_speed=20
    fadein=1 # fade in AND out
    fadein_speed=20
}

dwindle {
    pseudotile = yes 
    preserve_split = yes 
}

misc { 
  vrr = 1
}

master {
    new_is_master = true
}


input {
    kb_layout = br
    kb_variant = abnt2
    accel_profile = flat
    force_no_accel = false
    sensitivity = 0
}
```

**Comment by RochaLAJ** [comment]
> what if you would uuuuuuuuuuuuuuuuuhhhhhhhhhhhhhhhhhhhh
> 
> [patch.txt](https://github.com/hyprwm/Hyprland/files/10801816/patch.txt) this

Fixed :) 

After this change, i compiled by myself following recommendations available here: https://wiki.hyprland.org/Getting-Started/Installation/ 

Restarted my PC
Started Guild Wars 2. No problem
Started FFXIV. No problem. 

Played each for 30mins, this time seem to be working flawless.
Thank you for your time and effort! 

All the best
Luiz
[devices.txt](https://github.com/hyprwm/Hyprland/files/10808407/devices.txt)

**Comment by vaxerski** [comment]
fixed in 2bf7f9e4136ebef6db1159c636b163ecb96f23cc

**Comment by fliperama00** [comment]
I just noticed this bug now using Hyprland as my daily driver. Is it expected after edd0a14?
